### PR TITLE
[Sprint: 58] XD-3493: Update SI, Spring Core, and AMQP dependencies

### PR DESCRIPTION
**Pull request by dturanski**
This Now upgrades dependencies to Spring 4.2.1.RELEASE, Spring Integration 4.2.0.RELEASE, Spring Amqp 1.5.0.RELEASE. Also common dependencies with Boot 1.3.M5 are upgraded to the same versions where possible. Additionally there are some minor code changes to fix deprecated code, etc. This builds and shell script smoke test looks good.  The upgrade to Boot 1.3 was not done due to compatibility issues described below.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,6 +15,7 @@ target/
 .classpath
 .project
 .springBeans
+.cache-main
```
Comment (dturanski) [inline .gitignore]: the scala compiler leaves this behind now..

File: spring-xd-dirt/src/main/java/org/springframework/xd/dirt/integration/bus/converter/ByteArrayToStringMessageConverter.java [hunk 2]
```diff
@@ -65,7 +66,7 @@ public ByteArrayToStringMessageConverter() {
 	 * Don't need to manipulate message headers. Just return payload
 	 */
 	@Override
-	public Object convertFromInternal(Message<?> message, Class<?> targetClass) {
```
Comment (dturanski) [inline spring-xd-dirt/src/main/java/org/springframework/xd/dirt/integration/bus/converter/ByteArrayToStringMessageConverter.java]: deprecated in spring 4.2.1

File: spring-xd-dirt/src/main/java/org/springframework/xd/dirt/integration/bus/converter/ByteArrayToStringMessageConverter.java [hunk 3]
```diff
@@ -65,7 +66,7 @@ public ByteArrayToStringMessageConverter() {
...
+	protected Object convertFromInternal(Message<?> message, Class<?> targetClass, Object conversionHint) {
```
Comment (dturanski) [inline spring-xd-dirt/src/main/java/org/springframework/xd/dirt/integration/bus/converter/ByteArrayToStringMessageConverter.java]: I followed the same pattern used for core spring subclasses of AMC. The third argument is never used in these.

File: spring-xd-dirt/src/main/java/org/springframework/xd/dirt/server/options/SingleNodeOptions.java [hunk 4]
```diff
@@ -63,6 +61,19 @@ public String getXD_TRANSPORT() {
...
+
+
+	public void setAnalytics(String analytics) {
+		this.analytics = analytics;
+	}
+
+	public void setTransport(String transport) {
+		this.transport = transport;
+	}
+
```
Comment (dturanski) [inline spring-xd-dirt/src/main/java/org/springframework/xd/dirt/server/options/SingleNodeOptions.java]: Alternate setters required with boot 1.3, related to Relaxed binding. Not sure if we should remove the old ones @ericbottard

File: spring-xd-dirt/src/test/java/org/springframework/xd/dirt/security/SingleNodeApplicationWithSslTest.java [hunk 5]
```diff
@@ -21,9 +21,9 @@
 
 import org.apache.http.client.HttpClient;
 import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
-import org.apache.http.conn.ssl.SSLContextBuilder;
```
Comment (dturanski) [inline spring-xd-dirt/src/test/java/org/springframework/xd/dirt/security/SingleNodeApplicationWithSslTest.java]: deprecated

File: spring-xd-dirt/src/test/java/org/springframework/xd/dirt/plugins/job/support/StepExecutionJacksonMixInTests.java [hunk 6]
```diff
@@ -63,8 +63,8 @@ public void testSerializationOfSingleStepExecution() throws JsonProcessingExcept
 
 		final ObjectMapper objectMapper = new ObjectMapper();
 
-		objectMapper.addMixInAnnotations(StepExecution.class, StepExecutionJacksonMixIn.class);
-		objectMapper.addMixInAnnotations(ExecutionContext.class, ExecutionContextJacksonMixIn.class);
```
Comment (dturanski) [inline spring-xd-dirt/src/test/java/org/springframework/xd/dirt/plugins/job/support/StepExecutionJacksonMixInTests.java]: deprecated

File: spring-xd-module/src/main/java/org/springframework/xd/module/options/EnvironmentAwareModuleOptionsMetadataResolver.java [hunk 7]
```diff
@@ -216,7 +216,8 @@ private String fullyQualifiedKey(ModuleDefinition moduleDefinition, String optio
 	private ConfigurableEnvironment loadPropertySources(final String searchLocation, final String baseName) {
 		final ConfigurableEnvironment environment = new StandardEnvironment();
 		environment.merge(parentEnvironment);
-		new ConfigFileApplicationListener() {
```
Comment (dturanski) [inline spring-xd-module/src/main/java/org/springframework/xd/module/options/EnvironmentAwareModuleOptionsMetadataResolver.java]: class removed from boot 1.3

File: spring-xd-shell/src/test/java/org/springframework/xd/shell/security/SecuredShellAccessWithSslTests.java [hunk 8]
```diff
@@ -25,10 +25,10 @@
 import org.apache.http.auth.AuthScope;
 import org.apache.http.auth.UsernamePasswordCredentials;
 import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
-import org.apache.http.conn.ssl.SSLContextBuilder;
```
Comment (dturanski) [inline spring-xd-shell/src/test/java/org/springframework/xd/shell/security/SecuredShellAccessWithSslTests.java]: deprecated

File: spring-xd-ui/src/main/java/org/springframework/xd/dirt/web/config/SecurityConfiguration.java [hunk 9]
```diff
@@ -55,7 +55,7 @@
  */
 @Configuration
 @ConditionalOnProperty("security.basic.enabled")
-@EnableWebMvcSecurity
```
Comment (dturanski) [inline spring-xd-ui/src/main/java/org/springframework/xd/dirt/web/config/SecurityConfiguration.java]: deprecated

File: spring-xd-shell/src/test/java/org/springframework/xd/shell/util/TestSftpServer.java [hunk 10]
```diff
@@ -208,6 +208,7 @@ public DefaultSftpSessionFactory getSessionFactory() {
 		factory.setPort(this.port);
 		factory.setUser("foo");
 		factory.setPassword("foo");
+		factory.setAllowUnknownKeys(true);
```
Comment (dturanski) [inline spring-xd-shell/src/test/java/org/springframework/xd/shell/util/TestSftpServer.java]: see https://github.com/spring-projects/spring-xd/pull/1768

File: spring-xd-ui/src/main/java/org/springframework/xd/dirt/web/config/SecurityConfiguration.java [hunk 11]
```diff
@@ -96,31 +96,19 @@ protected void configure(HttpSecurity http) throws Exception {
...
-						.authorizeRequests()
-						.antMatchers("/").authenticated()
-						.antMatchers("/admin-ui/**").permitAll()
-						.antMatchers("/authenticate").permitAll()
-						.antMatchers("/security/info").permitAll()
-						.antMatchers("/assets/**").permitAll();
+		ExpressionUrlAuthorizationConfigurer<HttpSecurity>.ExpressionInterceptUrlRegistry security = http.csrf().disable().authorizeRequests().antMatchers(
+				"/").authenticated().antMatchers("/admin-ui/**").permitAll().antMatchers(
+						"/authenticate").permitAll().antMatchers("/security/info").permitAll().antMatchers(
+								"/assets/**").permitAll();
```
Comment (dturanski) [inline spring-xd-ui/src/main/java/org/springframework/xd/dirt/web/config/SecurityConfiguration.java]: yuck... new IDE, will fix

File: spring-xd-analytics/src/main/java/org/springframework/xd/analytics/metrics/integration/FieldValueCounterHandler.java [hunk 12]
```diff
@@ -68,7 +68,7 @@ public FieldValueCounterHandler(FieldValueCounterRepository fieldValueCounterRep
 				payload = jsonToTupleTransformer.transformPayload(payload.toString());
 			}
 			catch (Exception e) {
-				throw new MessageTransformationException(message, e);
```
Comment (dturanski) [inline spring-xd-analytics/src/main/java/org/springframework/xd/analytics/metrics/integration/FieldValueCounterHandler.java]: deprecated

**Comment by dturanski** [comment]
DO NOT MERGE yet, still working on an issue related to command line options with boot 1.3

**Comment by dturanski** [comment]
@ericbottard The latest commit works a bit better but I hit a nasty issue related to the spring-hateos version. xd-container will not start with spring-hateos-0.17.0 - it requires 0.19.0 (the declared dependency in boot 3.1.M5). However I had to downgrade to 0.17.0 to get around https://github.com/olivergierke/spring-restbucks/issues/37 (see note in dependencies.properties for more info). Other than that, hsqldb-server wont start as it is looking for XD_JMX_ENABLED. I haven't figured out where that is referenced @ilayaperumalg . Property settings for xd-singlenode and xd-admin seem to work as expected with this commit but will require more testing.  But the spring-hateos thing will be a challenge.

**Comment by garyrussell** [comment]
Merged as 496f0303d3fa7a482ffc7c30f6132a7cef9a5ce9

**Pull request closed without merging**